### ISSUE TITLE
Respect single Paragraphs type permissions in modal dialog

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/paragraphs#156]
As far as I can tell, the (experimental) modal dialog on the front end is only available for people with full permissions for the respective Paragraphs type. E.g., if I don't have the permission to _delete_ content of a certain Paragraphs type, I don't have access to the modal dialog at all (and can't _edit_ the Paragraph via the dialog).

I'd prefer a behavior where the dialog reflects the single Paragraphs type permissions, so that I have access to the dialog and can choose only actions which are allowed for my role. Example: I can't choose the "Remove" button but I'm still able to edit the Paragraph.

**Comment by laryn** [comment]
@olafgrabienski I agree with your preference but haven't worked on it recently enough to say how big of a change will be required. I know there is already some level of checking happening (e.g. if un/publishing is allowed, show that in the dropdown, otherwise not) but it may need further tweaking.
